pallets/gear/src/benchmarking/mod.rs

```diff
@@ -184,33 +186,73 @@ fn default_processor_context<T: Config>() -> ProcessorContext {
     }
 }
 
-fn verify_process(notes: Vec<JournalNote>) {
+fn verify_process((notes, err_len_ptrs): (Vec<JournalNote>, Vec<u32>)) {
     assert!(
         !notes.is_empty(),
         "Journal notes cannot be empty after execution"
     );
+
+    let mut pages_data = BTreeMap::new();
+
     for note in notes {
-        if let JournalNote::MessageDispatched { outcome, .. } = note {
-            match outcome {
-                DispatchOutcome::InitFailure { .. } | DispatchOutcome::MessageTrap { .. } => {
-                    panic!("Process was not successful")
-                }
-                _ => {}
+        match note {
+            JournalNote::MessageDispatched {
+                outcome: DispatchOutcome::InitFailure { .. } | DispatchOutcome::MessageTrap { .. },
+                ..
+            } => {
+                panic!("Process was not successful")
+            }
+            JournalNote::UpdatePage {
+                page_number, data, ..
+            } => {
+                pages_data.insert(page_number, data);
             }
+            _ => {}
         }
     }
+
+    for err_len_ptr in err_len_ptrs {
```

Comment:
Just advice: because error pointers lays in continuous interval, we can just pass as argument interval begin and size, and check that each byte is null.